Add docs about skip and debug tests suites in CI (#3550)
owncloud/docs: modules/developer_manual/pages/testing/acceptance-tests.adoc

For more information on Behat, and how to write acceptance tests using it, see http://behat.org/en/latest/guides.html[the Behat documentation].
For background information on Behaviour-Driven Development (BDD), see https://dannorth.net/whats-in-a-story/[Dan North resources].

== Skipping and Debugging Test Suites in CI

=== Skip Pipelines

> For various purposes, we may skip one or more CI pipelines. Use `skip` in drone config to skip the test pipelines. `skip` is available for `javascript`, `unit` and `acceptance` tests.

Reviewer: ```suggestion
For various purposes, we may skip one or more CI pipelines. Use `skip` in the drone config to skip the test pipelines. `skip` is available for `javascript`, `phpunit` and `acceptance` tests.
```

> For various purposes, we may skip one or more CI pipelines. Use `skip` in the drone config to skip the test pipelines. `skip` is available for `javascript`, `phpunit` and `acceptance` tests.

Reviewer: ```suggestion
For various purposes, you may skip one or more CI pipelines. Use `skip` in the drone config to skip the test pipelines. `skip` is available for `javascript`, `phpunit` and `acceptance` tests.
```

Usage:
[source,py]
----
...
'phpunit': {
    'allDatabases' : {
      'phpVersions': [
        '7.2',
      ],
      'skip': True
  },
}
...
'acceptance': {
  'api': {
    'suites': [
      'apiAuth',
      'apiAuthOcs',
      'apiAuthWebDav',
      'apiCapabilities',
      'apiComments'
    ],
    'skip': True
  },
}
...
----

=== Debug Specific Test Suites

> In CI, we may want to run only one or specific test suites for debugging purpose. To do so we can use `debugSuites` in drone config which takes list of suite names. If `debugSuites` is included with one or more test suites in it then only those suites will run in CI.

Reviewer: ```suggestion
In CI, we may want to run only one or specific test suites for debugging purposes. To do so we can use `debugSuites` in drone config which takes a list of suite names. If `debugSuites` is included with one or more test suites in it then only those suites will run in CI.
```

> In CI, we may want to run only one or specific test suites for debugging purposes. To do so we can use `debugSuites` in the drone config which takes a list of suite names. If `debugSuites` is included with one or more test suites in it then only those suites will run in CI.

Reviewer: ```suggestion
In CI, you may want to run only one or specific test suites for debugging purposes. To do so you can use `debugSuites` in the drone config which takes a list of suite names. If `debugSuites` is included with one or more test suites in it then only those suites will run in CI.
```

> (Note: remember to check `'skip': False` if you are using `skip`)

Reviewer: ```suggestion
(Note: remember to set `'skip': False` if you are using `skip`)
```

Usage:
[source,py]
----
...
'acceptance': {
  'api': {
    'suites': [
      'apiAuth',
      'apiAuthOcs',
      'apiAuthWebDav',
      'apiCapabilities',
      'apiComments'
    ],
    'debugSuites': ['apiAuth']
  }
}
...
----

> Similarly, in the case of test suites that run in parts, we can use `skipExceptParts` to specify only which part we want to run.

Reviewer: ```suggestion
Similarly, in the case of test suites that run in parts, we can use `skipExceptParts` to specify only which part(s) we want to run.
```

> Similarly, in the case of test suites that run in parts, we can use `skipExceptParts` to specify only which part(s) we want to run.

Reviewer: ```suggestion
Similarly, in the case of test suites that run in parts, you can use `skipExceptParts` to specify only which part(s) you want to run.
```

Usage:
[source,py]
----
...
'acceptance': {
  'apiProxy': {
    'suites': {
      'apiProxySmoketest': 'apiProxySmoke',
    },
    'numberOfParts': 8,

>     'skipExceptParts': [3]

Reviewer: ```suggestion
    'skipExceptParts': [3,7]
```

It might be useful to have more than one part in the example.
